joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -223,6 +215,51 @@ trait TypedScope[M <: MethodLike, F <: FieldLike, T <: TypeLike[M, F]](summary:
 
 }
 
+trait OverloadableScope[M <: OverloadableMethod] {
+  this: TypedScope[M, ?, ?] =>
+  override def tryResolveMethodInvocation(
+    callName: String,
+    argTypes: List[String],
+    typeFullName: Option[String] = None
+  )(implicit tag: ClassTag[M]): Option[M] = typeFullName match {
+    case None =>
+      // TODO: The typesInScope part is to imprecisely solve the unimplemented polymorphism limitation
+      membersInScope.collectFirst(matchingM(callName)).orElse {
+        typesInScope.flatMap(_.methods).collectFirst(matchingM(callName))
+      }
+    case Some(tfn) =>
+      val methodsWithEqualArgs = tryResolveTypeReference(tfn).flatMap { t =>
+        Option(
+          t.methods.filter(m => m.name == callName && m.parameterTypes.filterNot(_._1 == "this").size == argTypes.size)
```

Comment:
I feel we should use `isOverloadedBy` here, I think this solution mitigates `this` since `argTypes` doesn't include the receiver when used in `CSharp2Cpg`'s AstCreator. If possible, it would be a nice fix here but might be out of scope if too large
```suggestion
          t.methods.filter(m => m.name == callName && isOverloadedBy(m, argTypes))
```

Reply:
This causes some C# tests to fall over, added TODO comment for now.